ui/src/components/WMWorkerPoolEditor/index.jsx

```diff
@@ -377,13 +415,26 @@ export default class WMWorkerPoolEditor extends Component {
                 requiresAuth
                 tooltipOpen
                 icon={<DeleteIcon />}
-                onClick={this.handleOnClick}
+                onClick={onDialogActionOpen}
                 tooltipTitle="Delete"
                 className={classes.deleteIcon}
                 disabled={actionLoading}
               />
             </SpeedDial>
           )}
+          {dialogOpen && (
+            <DialogAction
+              open={dialogOpen}
+              onSubmit={this.handleOnClick}
```

Comment:
> @helfi92 on deleting worker pool, it shows Cannot read property 'currentTarget' of undefined. Also I haven't changed anything related to this. What should I do?

It is because you are calling `this.handleOnClick` here. The callback in `onSubmit` does not get an event but `handleOnClick` is trying to do `event.currentTarget`. See how other dialog does their `onSubmit`s.

Reply:
No other dialogs are doing this thing. All of them has separate functions for every action. I can't get it how to pass an while object in the dialog action.

Reply:
Try the following diff:

```diff
diff --git a/ui/src/components/WMWorkerPoolEditor/index.jsx b/ui/src/components/WMWorkerPoolEditor/index.jsx
index 925871408..dfe763ae8 100644
--- a/ui/src/components/WMWorkerPoolEditor/index.jsx
+++ b/ui/src/components/WMWorkerPoolEditor/index.jsx
@@ -274,6 +274,15 @@ export default class WMWorkerPoolEditor extends Component {
     }
   };
 
+  handleDeleteWorkerPool = () => {
+    const { workerPoolId1, workerPoolId2, ...payload } = this.state.workerPool;
+
+    return this.props.deleteRequest({
+      workerPoolId: joinWorkerPoolId(workerPoolId1, workerPoolId2),
+      payload,
+    });
+  };
+
   render() {
     const {
       classes,
@@ -411,7 +420,6 @@ export default class WMWorkerPoolEditor extends Component {
           {!isNewWorkerPool && (
             <SpeedDial>
               <SpeedDialAction
-                name="deleteRequest"
                 requiresAuth
                 tooltipOpen
                 icon={<DeleteIcon />}
@@ -425,7 +433,7 @@ export default class WMWorkerPoolEditor extends Component {
           {dialogOpen && (
             <DialogAction
               open={dialogOpen}
-              onSubmit={this.handleOnClick}
+              onSubmit={this.handleDeleteWorkerPool}
               onComplete={onDialogActionComplete}
               onClose={onDialogActionClose}
               onError={onDialogActionError}
@@ -433,7 +441,12 @@ export default class WMWorkerPoolEditor extends Component {
               title="Delete Worker Pool?"
               body={
                 <Typography>
-                  This will delete the Worker Pool {workerPool.workerPoolId1}.
+                  This will delete the worker pool{' '}
+                  {joinWorkerPoolId(
+                    workerPool.workerPoolId1,
+                    workerPool.workerPoolId2
+                  )}
+                  .
                 </Typography>
               }
               confirmText="Delete Worker Pool"

```